plugins/hive-plugin/src/main/java/edu/berkeley/ground/plugins/hive/GroundStore.java

```diff
@@ -75,21 +94,45 @@ public void rollbackTransaction() {
     }
 
     public void createDatabase(Database db) throws InvalidObjectException, MetaException {
-        GraphFactory graph = ground.getGraphFactory();
+        NodeFactory nf = getGround().getNodeFactory();
+        NodeVersionFactory nvf = getGround().getNodeVersionFactory();
+
+        Database dbCopy = db.deepCopy();
         try {
-            Graph g = graph.create(db.getName());
+            edu.berkeley.ground.api.versions.Type dbType = edu.berkeley.ground.api.versions.Type.fromString("string");
+            Tag dbTag = new Tag(null, dbCopy.getName(), Optional.of(dbCopy), Optional.of(dbType)); // fix
+            Optional<String> reference = Optional.of(dbCopy.getLocationUri());
+            Optional<String> versionId = Optional.empty();
+            Optional<String> parentId = Optional.empty(); // fix
+            HashMap<String, Tag> tags = new HashMap<>();
+            tags.put(dbCopy.getName(), dbTag);
+            Optional<Map<String, Tag>> tagsMap = Optional.of(tags);
+            Optional<Map<String, String>> parameters = Optional.of(dbCopy.getParameters());
+            String name = HiveStringUtils.normalizeIdentifier(dbCopy.getName());
+            String nodeId = nf.create(name).getId();
+            NodeVersion n = nvf.create(tagsMap, versionId, reference, parameters, nodeId, parentId);
+            dbMap.put(name, n.getId());
         } catch (GroundException e) {
+            LOG.error("error creating database " + e);
+            throw new MetaException(e.getMessage());
         }
-
     }
 
     public Database getDatabase(String name) throws NoSuchObjectException {
-        // TODO Auto-generated method stub
-        return null;
+        NodeVersion n;
+        try {
+            n = getGround().getNodeVersionFactory().retrieveFromDatabase(name);
```

Comment:
The is the signature of `NodeVersionFactory#retrieveFromDatabase`is `public abstract NodeVersion retrieveFromDatabase(String id) throws GroundException;`You should be passing in the `id` of the `NodeVersion`, not the `name`.